wallet-new/test/unit/WalletUnitTest.hs

```diff
@@ -105,15 +106,19 @@ testPureWallet = do
       valid "simple and incremental empty wallets are equivalent" $
         walletEquivalent specEmpty incrEmpty WalletEmpty
 
-    describe "Inductive Wallet Generation" $ do
-      prop "constructed inductive wallet satisfies invariants" $ do
-        forAll genInductive $ \wallet -> conjoin
-          [ isValidated $ walletInvariants specEmpty wallet
-          , isValidated $ walletInvariants incrEmpty wallet
-          , isValidated $ walletInvariants rollEmpty wallet
-          , isValidated $ walletEquivalent specEmpty incrEmpty WalletEmpty
-          , isValidated $ walletEquivalent specEmpty rollEmpty WalletEmpty
-          ]
+    beforeAll (generate genInductive) $ do
+      describe "Inductive wallet generations satisfy invariants" $ do
+        it "specEmpty " $
+          shouldBeValidated . walletInvariants specEmpty
+        it "incrEmpty" $
+          shouldBeValidated . walletInvariants incrEmpty
+        it "rollEmpty" $
+          shouldBeValidated . walletInvariants rollEmpty
+      describe "Wallets are equivalent after interpretation" $ do
+        it "spec and incr" $
+          shouldBeValidated . walletEquivalent specEmpty incrEmpty
+        it "spec and roll" $
+          shouldBeValidated . walletEquivalent specEmpty rollEmpty
```

Comment:
hspec's `beforeAll` runs the IO action once and the result is made available to all the `it`'s beforehand. These are point-free, but it looks like `it "foo" $ \wallet -> shouldBeValidated ...` when eta expanded